aip/general/0162.md

```diff
@@ -231,178 +200,80 @@ there are two potential variants:
-- The method **should** return the resource, and the resource name **must**
-  include the revision ID.
-- The request message **must** include the `name` field.
-  - The field **should** be [annotated as required][aip-203].
-  - The field **should** identify the [resource type][aip-123] that it
-    references.
+Revisions being resources under nested collection make revisions a first class
+citizen.
 
-### Rollback
+- Revisions can offer standard get, list, and delete methods.
+- It retains the flexibility of extending new fields to revision in addition to
+  the resource message.
 
-A common use case for a resource with a revision history is the ability to roll
-back to a given revision. APIs **should** handle this with a `Rollback` custom
-method:
+### Consolidation of tagging use cases
 
-```proto
-rpc RollbackBook(RollbackBookRequest) returns (Book) {
-  option (google.api.http) = {
-    post: "/v1/{name=publishers/*/books/*}:rollback"
-    body: "*"
-  };
-}
-```
+Previously, committing and tagging a resource were two separate methods on a
+resource, enabling the following user journeys:
 
-- The method **must** use the `POST` HTTP verb.
-- The method **should** return the resource, and the resource name **must**
-  include the revision ID.
+- Commit
+  - Creating a revision with the current resource configuration.
+- Tag
+  - Taking an existing revision and giving it an alias.
 
-```proto
-message RollbackBookRequest {
-  // The book being rolled back.
-  string name = 1 [
-    (google.api.field_behavior) = REQUIRED,
-    (google.api.resource_reference) = {
-      type: "library.googleapis.com/Book"
-    }];
+Tagging is effectively giving a second resource ID to a revision.
 
-  // The revision ID to roll back to.
-  // It must be a revision of the same book.
-  //
-  //   Example: c7cfa2a8
-  string revision_id = 2 [(google.api.field_behavior) = REQUIRED];
-}
-```
+## History
 
-- The request message **must** have a `name` field to identify the resource
-  being rolled back.
-  - The field **should** be [annotated as required][aip-203].
-  - The field **should** identify the [resource type][aip-123] that it
-    references.
-- The request message **must** include a `revision_id` field.
-  - The API **must** fail the request with `NOT_FOUND` if the revision does not
-    exist on that resource.
-  - The field **should** be [annotated as required][aip-203].
+### Switching from from a collection extension to a subcollection
 
-**Note:** When rolling back, the API should return a _new_ revision of the
-resource with a _new_ revision ID, rather than reusing the original ID. This
-avoids problems with representing the same revision being active for multiple
-ranges of time.
+In 2023-09, revisions are abstracted as nested resource. Prior to this,
+revisions are more like extension of an existing resource by using `@` symbol.
+List and delete revisions are custom method on the resource collection. Getting
```

Comment:
custom method => custom methods

(And possibly "Operations to list and delete revisions"? (Getting really nitpicky here...)